ansible-automation-platform-operator/tasks/main.yml

```diff
@@ -0,0 +1,115 @@
+---
+- name: Setting Up Results Var
+  ansible.builtin.set_fact:
+    assessment_result: "pass"
+# 1. Testing AutomationController
+- name: Testing AutomationController
+  block:
+    - name: Create AutomationController
+      kubernetes.core.k8s:
+        name: aap-test
+        namespace: "{{ target_namespace }}"
+        api_version: automationcontroller.ansible.com/v1beta1
+        kind: AutomationController
+        definition:
+          metadata:
+            labels:
+              opdev: opcap
+          spec:
+            create_preload_data: true
+            route_tls_termination_mechanism: Edge
+            garbage_collect_secrets: false
+            ingress_type: Route
+            loadbalancer_port: 80
+            no_log: false
+            image_pull_policy: IfNotPresent
+            projects_storage_size: 8Gi
+            auto_upgrade: true
+            task_privileged: false
+            projects_storage_access_mode: ReadWriteMany
+            set_self_labels: true
+            projects_persistence: false
+            replicas: 1
+            admin_user: admin
+            loadbalancer_protocol: http
+            nodeport_port: 30080
+        state: present
+    - name: Test AutomationController conditions type Running
+      kubernetes.core.k8s_info:
+        namespace: "{{ target_namespace }}"
+        name: aap-test
+        api_version: automationcontroller.ansible.com/v1beta1
+        kind: AutomationController
+        wait: true
+        wait_condition:
+          status: "True"
+          type: "Running"
+        wait_timeout: 300
+      register: automationcontroller
+      failed_when: (automationcontroller.resources[0].status.conditions | selectattr('type', 'search', 'Running') | list | last).status != "True"
+    - name: Test AutomationController conditions type Failure
+      kubernetes.core.k8s_info:
+        namespace: "{{ target_namespace }}"
+        name: aap-test
+        api_version: automationcontroller.ansible.com/v1beta1
+        kind: AutomationController
+        wait: true
+        wait_condition:
+          status: "False"
+          type: "Failure"
+        wait_timeout: 300
+      register: automationcontroller
+      failed_when: (automationcontroller.resources[0].status.conditions | selectattr('type', 'search', 'Failure') | list | last).status != "False"
+    - name: Test AutomationController conditions type Successful
```

Comment:
Yes. all tasks work fine and pass on my 3 controller/3 worker default ocp cluster on aws.

Reply:
>> But are you seeing them just sit there for 5 minutes?

def not...all these verification tasks are met...the first one within 30 sec or so....the second one shortly after....the third one takes a few min on my cluster.

Reply:
The first one goes pretty quickly. But from what I'm reading in the tasks, and what I see in the execution, is that the first task creates the CR. Then, the subsequent tasks are looking at the same resource to see if various conditions are present. However, what happens, if you specify "wait: true", and the wait_condition, it will literally wait until those conditions are met. The first k8s_info task usually succeeds quickly, because it's the happy path. The next two are (I presume) error states, and are not likely to occur. So what I witnessed is that the task waits for 5 minutes but if those are not met (again, not likely) the tasks will just timeout after 5 minutes. I'll try again to make sure that's what I'm seeing.

Reply:
>> it will literally wait until those conditions are met.

That's the way it should behave. The next two aren't error states. They are what is needed to signal a successful task.

Reply:
I think I was looking at it the wrong way. Thanks for the clarification.